protocol/consistencychecks.go

```diff
@@ -45,18 +43,48 @@ func NewCC(savedSTR *DirSTR, useTBs bool, signKey sign.PublicKey) *ConsistencyCh
 	if !useTBs {
 		panic("[coniks] Currently the server is forced to use TBs")
 	}
+	aud := NewAuditor(signKey, savedSTR)
 	cc := &ConsistencyChecks{
-		SavedSTR: savedSTR,
-		Bindings: make(map[string][]byte),
-		useTBs:   useTBs,
-		signKey:  signKey,
+		aud,
+		make(map[string][]byte),
+		useTBs,
+		nil,
 	}
 	if useTBs {
 		cc.TBs = make(map[string]*TemporaryBinding)
 	}
 	return cc
 }
 
+// CheckEquivocation checks for possible equivocation between
+// an auditors' observed STRs and the client's own view.
+// CheckEquivocation() first verifies the STR range received
+// in msg if msg contains more than 1 STR, and
+// then checks the most recent STR in msg against
+// the cc.verifiedSTR.
+// CheckEquivocation() is called when a client receives a response to a
+// message.AuditingRequest from an auditor.
+func (cc *ConsistencyChecks) CheckEquivocation(msg *Response) error {
```

Comment:
IMHO, we should open a separate pull for this feature, since I think we should cover something like the number of chosen auditors, and design a set of APIs for the `/protocol` level.